pilz_robot_programming/src/pilz_robot_programming/commands.py

```diff
@@ -217,40 +219,56 @@ def _cmd_to_request(self, robot):
-
-        # goal as list of int or float in joint space
-        elif isinstance(self._goal, list):
-            joint_names = robot._robot_commander.get_group(self._planning_group).get_active_joints()
-            joint_values = self._get_joint_pose(robot)
+        req.goal_constraints.append(Constraints())
 
-            if len(joint_names) != len(joint_values):
-                raise IndexError("Given joint goal does not match the planning group " + req.group_name + ".")
-
-            for joint_name, joint_value in zip(joint_names, joint_values):
-                joint_constraint = JointConstraint()
-                joint_constraint.joint_name = joint_name
-                joint_constraint.position = joint_value
-                joint_constraint.weight = 1
-                goal_constraints.joint_constraints.append(joint_constraint)
+        convertion_methods = [self._set_constraint_by_joint_values,
+                              self._set_constraint_by_pose,
+                              self._set_constraint_by_stamped_pose,
+                              self._set_constraint_by_joint_state]
 
+        for method in convertion_methods:
+            try:
+                method(req)
+                break
+            except (TypeError, AttributeError, AssertionError):
+                pass
         else:
             raise NotImplementedError("Unknown type of goal is given.")
 
-        req.goal_constraints.append(goal_constraints)
-
         return req
 
+    def _set_constraint_by_joint_values(self, req, joint_names=()):
+        assert not isinstance(self._goal, str)
+        joint_names = joint_names if len(joint_names) != 0 \
+            else self._robot._robot_commander.get_group(self._planning_group).get_active_joints()
+        joint_values = self._get_joint_pose()
+        if len(joint_names) != len(joint_values):
+            raise IndexError("Given joint goal does not match the planning group " + req.group_name + ".")
+        for joint_name, joint_value in zip(joint_names, joint_values):
+            joint_constraint = JointConstraint()
+            joint_constraint.joint_name = joint_name
+            joint_constraint.position = joint_value
+            joint_constraint.weight = 1
+            req.goal_constraints[0].joint_constraints.append(joint_constraint)
+
+    def _set_constraint_by_pose(self, req):
+        goal_pose = self._get_goal_pose()
+        robot_reference_frame = self._robot._robot_commander.get_planning_frame()
+        req.goal_constraints[0].orientation_constraints.append(
+            _to_ori_constraint(goal_pose, robot_reference_frame, self._target_link))
+        req.goal_constraints[0].position_constraints.append(
+            _to_pose_constraint(goal_pose, robot_reference_frame, self._target_link))
+
+    def _set_constraint_by_stamped_pose(self, req):
+        self._reference_frame = self._goal.header.frame_id if self._goal.header.frame_id != "" else _DEFAULT_BASE_LINK
```

Comment:
@agutenkunst suggested an assertion that the time in the header is zero. Otherwise users could try to set a timestamp in the future and expect the robot to move at a later point in time - which we currently ignore...